sdk/storage/azure-storage-common/inc/azure/storage/common/constants.hpp

```diff
@@ -4,20 +4,18 @@
 
 #pragma once
 
-namespace Azure { namespace Storage { namespace Details {
-  constexpr static const char* c_BlobServicePackageName = "storage-blobs";
-  constexpr static const char* c_DatalakeServicePackageName = "storage-files-datalake";
-  constexpr static const char* c_FileServicePackageName = "storage-files-shares";
-  constexpr static const char* c_QueueServicePackageName = "storage-queues";
-  constexpr static const char* c_HttpQuerySnapshot = "snapshot";
-  constexpr static const char* c_HttpQueryVersionId = "versionid";
-  constexpr static const char* c_StorageScope = "https://storage.azure.com/.default";
-  constexpr static const char* c_HttpHeaderDate = "date";
-  constexpr static const char* c_HttpHeaderXMsVersion = "x-ms-version";
-  constexpr static const char* c_HttpHeaderRequestId = "x-ms-request-id";
-  constexpr static const char* c_HttpHeaderClientRequestId = "x-ms-client-request-id";
-  constexpr static const char* c_HttpHeaderContentType = "content-type";
-  constexpr static const char* c_defaultSasVersion = "2020-02-10";
+#define c_BlobServicePackageName "storage-blobs"
+#define c_DatalakeServicePackageName "storage-files-datalake"
+#define c_FileServicePackageName "storage-files-shares"
+#define c_QueueServicePackageName "storage-queues"
+#define c_HttpQuerySnapshot "snapshot"
+#define c_HttpQueryVersionId "versionid"
+#define c_StorageScope "https://storage.azure.com/.default"
+#define c_HttpHeaderDate "date"
+#define c_HttpHeaderXMsVersion "x-ms-version"
```

Comment:
Out of curiosity, what's motivating this change?

Reply:
is there anything wrong with macro?

Reply:
Not that I am aware off.

Reply:
don't use MACRO,
those variables are not allocated in memory. Will they be always temporally allocated?  
Don't we pass strings by reference?   Then are you passing refs to temp objects?
https://rules.sonarsource.com/cpp/RSPEC-5028
https://stackoverflow.com/questions/42388077/constexpr-vs-macros

Why did you decide to make this change?
Can you get @CaseyCarter / @RickWinter  advise about this before merging?

Reply:
Macros are generally disliked due to lack of namespacing: your macro named `potato` stomps all over any other uses of the identifier `potato` in the program. These macros in particular are problematic in that they use names outside of the `ALL_CAPS_SNAKE_CASE` identifier space that is conventionally reserved for macros. I'd recommend avoiding macros for this use case, but if you must then I strongly suggest changing the names: we don't want to follow `<Windows.h>`'s example of defining `min`, `max`, and `small`.